data/gen_graphs.py

```diff
@@ -0,0 +1,88 @@
+import random
+import networkx as nx
+import subprocess
+
+
+def gen_erdos_renyi_graph_single_component(n, m):
+    if_connected = 0
+    for i in range(1000):
+        G = nx.gnm_random_graph(n, m)
+        if_connected = nx.number_connected_components(G)
+        if if_connected == 1:
+            break
+    if if_connected == 1:
+        return nx.to_dict_of_lists(G)
+    else:
+        return None
+
+
+for i in range(300):
+    n_nodes_ran = 1000
+
+    # minimum number of edges for undirected connected graph is (n-1) edges
+    # maximal number of edges for undirected connected graph is n(n-1)/2 edges
+    edge_numbers = n_nodes_ran * (n_nodes_ran - 1) / 2 - n_nodes_ran
+
+    n_edges_very_sparse = random.randint(n_nodes_ran, n_nodes_ran + edge_numbers / 4)
+    n_edges_sparse = random.randint(
+        n_nodes_ran + edge_numbers / 4, n_nodes_ran + edge_numbers / 2
+    )
+    n_edges_dense = random.randint(
+        n_nodes_ran + edge_numbers / 2, n_nodes_ran + edge_numbers * 3 / 4
+    )
+    n_edges_very_dense = random.randint(
+        n_nodes_ran + edge_numbers * 3 / 4, n_nodes_ran + edge_numbers
+    )
+
+    very_sparse = gen_erdos_renyi_graph_single_component(
+        n_nodes_ran, n_edges_very_sparse
+    )
```

Comment:
This also, without seed, people will get different numbers of edge. Is this intended?

Reply:
Fixed random seed.